### PR TITLE
Task removal

### PR DESCRIPTION
**Pull request by vampcat**
Removes "Tasks" from Rendering code, to make the code simpler.

Testing: Run the game, ensure no exceptions come up, and that everything draws perfectly.

### CODE REVIEW

**Comment by vampcat** [comment]
Tagging @emanuele3d for review!

**Comment by GooeyHub** [comment]
Uh oh, something went wrong with the build. Need to check on that

**Comment by GooeyHub** [comment]
Uh oh, something went wrong with the build. Need to check on that

**Comment by GooeyHub** [comment]
Uh oh, something went wrong with the build. Need to check on that

**Comment by GooeyHub** [comment]
Uh oh, something went wrong with the build. Need to check on that

**Comment by GooeyHub** [comment]
Hooray Jenkins reported success with all tests good!

File: engine-tests/src/test/java/org/terasology/rendering/dag/RenderTaskListGeneratorTest.java [hunk 1]
```diff
@@ -36,13 +36,13 @@ public void testSimpleReducePersistingStateChanges() {
         List<RenderPipelineTask> taskList = renderTaskListGenerator.generateFrom(orderedNodes);
 
         assertEquals("----- AlphaNode",        taskList.get(0).toString().trim()); // Strictly speaking we don't need
-        assertEquals("SetNameTask: foo",       taskList.get(1).toString().trim()); // trimming MarkerTask.toString(),
-        assertEquals("AlphaNode: process()",   taskList.get(2).toString().trim()); // resulting in "----- <NodeName>"
+        assertEquals("SetName: foo",       taskList.get(1).toString().trim()); // trimming MarkerTask.toString(),
```
Comment (emanuele3d) [inline engine-tests/src/test/java/org/terasology/rendering/dag/RenderTaskListGeneratorTest.java]: Once you are happy with the changes you did to the actual code, please update the test code in its own commit.

File: engine-tests/src/test/java/org/terasology/rendering/dag/SetNameTask.java [hunk 2]
```diff
@@ -1,37 +0,0 @@
...
- * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
- * See the License for the specific language governing permissions and
- * limitations under the License.
- */
-package org.terasology.rendering.dag;
-
-/**
- * TODO: Add javadocs
- */
-public class SetNameTask implements RenderPipelineTask {
```
Comment (emanuele3d) [inline engine-tests/src/test/java/org/terasology/rendering/dag/SetNameTask.java]: Please execute the removal of all task classes in its own, separate commit.

File: engine/src/main/java/org/terasology/rendering/dag/AbstractNode.java [hunk 3]
```diff
@@ -38,7 +38,6 @@
     private Set<StateChange> desiredStateChanges = Sets.newLinkedHashSet();
     private Set<StateChange> desiredStateResets = Sets.newLinkedHashSet();
     private Map<ResourceUrn, BaseFBOsManager> fboUsages = Maps.newHashMap();
-    private NodeTask task;
```
Comment (emanuele3d) [inline engine/src/main/java/org/terasology/rendering/dag/AbstractNode.java]: Once the task classes have been removed, in the following commit you can remove/replace all lines that used that functionality.

File: engine/src/main/java/org/terasology/rendering/dag/nodes/AlphaRejectBlocksNode.java [hunk 4]
```diff
@@ -52,7 +52,7 @@
  * the color stored in the frame buffer and the resulting color overwrites the previously stored one.
  */
 public class AlphaRejectBlocksNode extends AbstractNode implements WireframeCapable {
-    private static final ResourceUrn CHUNK_SHADER = new ResourceUrn("engine:prog.chunk");
```
Comment (emanuele3d) [inline engine/src/main/java/org/terasology/rendering/dag/nodes/AlphaRejectBlocksNode.java]: 1. These are changes that are not relevant to the goal of this PR. It's ok to do them in this PR if they caught your attention but they should be in their own commit.
2. this URN, when used with getMaterial, returns a material, not a shader.

File: engine/src/main/java/org/terasology/rendering/dag/stateChanges/SetInputTexture.java [hunk 5]
```diff
@@ -37,62 +36,56 @@
...
+
+        this.material = getMaterial(materialUrn);
     }
 
-    private SetInputTexture(int textureSlot, ResourceUrn materialURN, String materialParameter) {
+    private SetInputTexture(int textureSlot, ResourceUrn materialUrn, String materialParameter) {
         this.textureSlot = textureSlot;
         this.textureId = 0;
-        this.materialURN = materialURN;
+        this.materialUrn = materialUrn;
```
Comment (emanuele3d) [inline engine/src/main/java/org/terasology/rendering/dag/stateChanges/SetInputTexture.java]: This kind of renaming is another change that can go in its own commit.
